CRM/MembershipExtras/Hook/PostProcess/UpdateSubscription.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+use CRM_MembershipExtras_Service_InstallmentReceiveDateCalculator as InstallmentReceiveDateCalculator;
+
+/**
+ * Post-processes Recurring Conribution Update form.
+ */
+class CRM_MembershipExtras_Hook_PostProcess_UpdateSubscription {
+
+  /**
+   * Form to be processed.
+   *
+   * @var \CRM_Contribute_Form_UpdateSubscription
+   */
+  private $form;
+
+  public function __construct(CRM_Contribute_Form_UpdateSubscription $form) {
+    $this->form = $form;
+    $this->recurringContributionID = CRM_Utils_Request::retrieve('crid', 'Integer', $this->form, TRUE);
+    $this->recurringContribution = civicrm_api3('ContributionRecur', 'getsingle', [
+      'id' => $this->recurringContributionID
+    ]);
+    $this->receiveDateCalculator = new InstallmentReceiveDateCalculator($this->recurringContribution);
+  }
+
+  /**
+   * Postprocesses CRM_Contribute_Form_UpdateSubscription to deal with added
+   * fields.
+   */
+  public function postProcess() {
+    $updateInstallments = CRM_Utils_Request::retrieve('update_installments', 'Integer', $this->form, FALSE);
+
+    if ($updateInstallments == '1') {

```

Comment:
I think using guard statement is better here : 

if (!$updateInstallments) {
return;
}